lib/bin/filter_clipped_alignments.py

```diff
@@ -0,0 +1,66 @@
+import csv
+import re
+import argparse
+
+front_pattern = re.compile(r"^\d+[SH]")
+end_pattern = re.compile(r"\d+[SH]$")
+
+
+def split_cigar(cigar_str):
+    """splits the count from type in a cigar string"""
+    return int(cigar_str[:-1]), cigar_str[-1]
+
+
+def get_clipping(cigar, alength, pattern):
+    """Get the amount of clipping from the front or end of the cigar string"""
+    clipping = 0
+
+    match = re.search(pattern, cigar)
+    if match:
+        clip_length, clip_type = split_cigar(match.group())
+        clipping += clip_length
+        if clip_type == "H":
+            alength += clip_length
+
+    return clipping, alength
+
+
+def main(reads_to_contig_sam, output_file, max_percent):
+    with open(reads_to_contig_sam, "r") as csv_file, open(
+        output_file, mode="w", newline=""
+    ) as output_file:
+        csv_reader = csv.reader(csv_file, delimiter="\t")
+        output = csv.writer(
+            output_file,
+            delimiter="\t",
+            lineterminator="\n",
+            quotechar="\t",
```

Comment:
this and `QUOTE_NONE` were necessary for the csv module to stop mangling the quality string in case of quotation marks `"`